Robots_diagram/src/main/webapp/app/diagram/controllers/DiagramMenuController.ts

```diff
@@ -0,0 +1,345 @@
+    private currentDiagramFolderId: string;
+    private canBeDeleted: boolean;
+
+    constructor($scope) {
+        this.diagramController = $scope.vm;
+        this.folderLevel = 0;
+        this.currentDiagramFolderId = "";
+        this.currentDiagramName = "";
+        this.canBeDeleted = false;
+
+        var user: string;
+        $.ajax({
+            type: 'POST',
+            url: 'getUser',
+            dataType: 'text',
+            success: function (response) {
+                user = response;
+            }
+        });
+        this.user = user;
+        this.currentFolderId = this.user + "root_0";
+
+        $.ajax({
+            type: 'POST',
+            url: 'createFolder',
+            dataType: 'text',
+            contentType: 'application/json',
+            data: (ExportManager.exportFolderToJSON(this.currentFolderId, "root", "")),
+            success: function () {},
+            error: function (response, status, error) {
+                console.log("error: " + status + " " + error);
+            }
+        });
+
+        var menuController = this;
+
+        $(document).ready(function() {
+            $('.modal-footer button').click(function() {
+                menuController.currentFolderId = menuController.user + "root_0";
+                menuController.folderLevel = 0;
+            });
+            $('#saveAfterCreate').click(function () {
+                menuController.canBeDeleted = true;
+                menuController.saveCurrentDiagram();
+            });
+        });
+    }
+
+    private clearScene(): void {
+        this.diagramController.graph.clear();
+        this.diagramController.nodesMap = {};
+        this.diagramController.linksMap = {};
+        $(".property").remove();
+        this.diagramController.currentElement = undefined;
+        this.canBeDeleted = false;
+    }
+
+    private clearAll(): void {
+        this.clearScene();
+        this.currentDiagramName = "";
+        this.currentDiagramFolderId = "";
+    }
+
+    private saveDiagram(diagramName: string): void {
+        var menuController = this;
+        this.currentDiagramName = diagramName;
+        this.currentDiagramFolderId = this.currentFolderId;
+        $.ajax({
+            type: 'POST',
+            url: 'saveDiagram',
+            dataType: 'text',
+            contentType: 'application/json',
+            data: (ExportManager.exportDiagramStateToJSON(diagramName, this.currentFolderId,
+                this.diagramController.nodesMap, this.diagramController.linksMap)),
+            success: function (response) {
+                if (response === "OK") {
+                    menuController.currentFolderId = menuController.user + "root_0";
+                    menuController.folderLevel = 0;
+                    $('#diagrams').modal('hide');
+
+                    if (menuController.canBeDeleted) {
+                        menuController.clearAll();
+                    }
+                }
+                else {
```

Comment:
Все-таки хочется, чтобы обработка ошибки была не в success, а в error.
Вот тут в пункте "What Is New Since Spring 3.X?" описано как сделать это через исключения: http://www.javacodegeeks.com/2012/11/spring-mvc-error-handling-example.html